hs_core/hydroshare/utils.py

```diff
@@ -754,6 +774,20 @@ def resource_file_add_process(resource, files, user, extract_metadata=False,
     return resource_file_objects
 
 
+def create_empty_contents_directory(resource):
+    if resource:
+        res_id = resource.short_id
+        if resource.resource_federation_path:
```

Comment:
I think it would be better to use a syntax like this so that we don't have to check where the resource is stored (federated or not) all the time:

```istorage = resource.get_istorage()```